api/src/main/java/org/openmrs/module/ugandaemrreports/definition/cohort/evaluator/CurrentlyInProgramCohortEvaluator.java

```diff
@@ -0,0 +1,66 @@
+package org.openmrs.module.ugandaemrreports.definition.cohort.evaluator;
+
+import org.openmrs.Cohort;
+import org.openmrs.PatientProgram;
+import org.openmrs.annotation.Handler;
+import org.openmrs.module.reporting.cohort.EvaluatedCohort;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.InProgramCohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.evaluator.CohortDefinitionEvaluator;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.querybuilder.HqlQueryBuilder;
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.openmrs.module.ugandaemrreports.definition.cohort.definition.CurrentlyInProgramCohortDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * This class was copied from the the InProgramCohortEvaluator in the openmrs-reporting module
```

Comment:
The implementation in this class was copied from the the InProgramCohortEvaluator, to cater for DSDM where the patients complete a program on the same day that they start a new one, therefore during the changeover date they are in actually in 2 programs on the same day. 

The change is in the last filter from TBD to whereGreaterOrNull so that the last program the patient was enrolled in is ignored